scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -198,6 +200,13 @@ pub struct TomlPackage {
     /// **UNSTABLE** This package does not depend on Cairo's `core`.
     pub no_core: Option<bool>,
     pub cairo_version: Option<MaybeWorkspaceField<VersionReq>>,
+    pub experimental: Option<MaybeWorkspaceField<ScarbExperimentalFeaturesConfig>>,
+}
+
+#[derive(Clone, Debug, Default, Hash, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct ScarbExperimentalFeaturesConfig {
+    pub negative_impls: Option<bool>,
 }
```

Comment:
My suggestion would be:
- Implement `From` trait for both `ScarbExperimentalFeaturesConfig` -> `ExperimentalFeaturesConfig` conversion and vice versa. 
- Use `ScarbExperimentalFeaturesConfig {  }` when converting to make compiler ensure we use all fields.
- Use `ScarbExperimentalFeaturesConfig ` in the `toml` structs and  `ExperimentalFeaturesConfig` everywhere else.
- Use `Into::into` to convert where needed.
- Hack: you can create second struct called `ExperimentalFeaturesConfig` (instead of calling it `ScarbExperimentalFeaturesConfig`) by enclosing it in some `mod { }`. E.g. 
```
mod serde_ext {
    struct ExperimentalFeaturesConfig {}
}
``` 
and use it by `serde_ext::ExperimentalFeaturesConfig`.

Reply:
Changed implementation to match [Marek's suggestion](https://github.com/software-mansion/scarb/pull/1011#discussion_r1440251944)